### DISCUSSION TITLE
The plugin does not notice new selection

### DISCUSSION THREAD

**Discussion by suedenjin** [coddingtonbear/obsidian-web#185] [Exotic Browser Support]
### Describe the bug

If I've used the plugin once on a page I have to refresh the page before I use it again. If I don't, it will send the same selection as the first time to Obsidian, resulting in a duplicate. 

Even if I navigate to a different part of the same website, the plugin does not notice, and keeps the first selection. I have to either refresh or go to a different website for the plugin to refresh. 

Is this expected behaviour? 

(Picture shows that the plugin is not showing current selection) 
![Screenshot 2024-04-01 at 21 25 09](https://github.com/coddingtonbear/obsidian-web/assets/165696316/95212aa5-d52c-42ba-b18d-33d9245a49a1)

### To Reproduce

1. Select some text
2. Use the plugin - text is successfully sent to Obsidian
3. Select a new section of text
4. Use the plugin again - the new section of text is not noticed by the plugin, instead the text selected in Step 1 is previewed and sent to Obsidian

### Expected Behavior

1. Select some text
2. Use the plugin - text is successfully sent to Obsidian
3. Select a new section of text
4. Use the plugin again - the plugin notices a new section of text has been selected and sends that instead

### Operating System

Mac 10.5.7 Catalina

### Browser

Brave

### Browser Version

1.64.113

**Comment by coddingtonbear** [comment]
I'm afraid I can't say much unless you upload a bug report export (see the extension options page in the upper-right corner).  Could you upload one so I have enough information to troubleshoot?

**Reply by suedenjin** [reply]
Thank you, sure! 
[obsidian-local-rest-api.bug_report.json](https://github.com/coddingtonbear/obsidian-web/files/14828189/obsidian-local-rest-api.bug_report.json)

**Comment by coddingtonbear** [comment]
So, I was pretty sure I was going to tell you "Oh, your version of Obsidian Web is just out-of-date" because a few months ago there was a bug around that behavior, but I do see that you're using the latest version.

I haven't personally been able to reproduce this issue; so I don't really have any good ideas.  What I can say is that this particular handler is the one responsible for refreshing context: https://github.com/coddingtonbear/obsidian-web/blob/39ac0f6b0aec12af82df7fc98d328b83a3d1af6f/src/popup.tsx#L429 -- it appears that we'll refresh it whenever the form is opened after having been closed, but maybe something else needs to be added to the dependency list.

**Comment by coddingtonbear** [comment]
I still haven't reproduced this, but I've made a speculative change that might maybe help?  I've submitted it to the chrome web store for review, and it'll probably begin rolling out over the next day or so.  It will be version 3.3.4 and once it's released (see 'Details' at https://chromewebstore.google.com/detail/obsidian-web/edoacekkjanmingkbkgjndndibhkegad), you can force your browser to update to the latest version by following these instructions: https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-ask-Chrome-to-update-extensions-immediately%3F.
